frontend/src/components/Layout/Header/HeaderContainer.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { useState } from 'react';
+
+import { User } from 'types/user';
+
+import Header from './Header';
+
+const HeaderContainer: React.FC<User> = ({ firstName, lastName, avatar }) => {
+  const [anchorElNotifications, setAnchorElNotifications] = useState<null | HTMLElement>(null);
+  const handleNotificationsOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorElNotifications(anchorElNotifications ? null : event.currentTarget);
+  };
+  const openNotifications = Boolean(anchorElNotifications);
+  const [anchorElFilter, setAnchorElFilter] = useState<null | HTMLElement>(null);
+  const handleFilterOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorElFilter(anchorElFilter ? null : event.currentTarget);
+  };
```

Comment:
All the states should be at the top of page, and only then functions.
```suggestion
  const [anchorElNotifications, setAnchorElNotifications] = useState<null | HTMLElement>(null);
  const [anchorElFilter, setAnchorElFilter] = useState<null | HTMLElement>(null);
  
  const handleNotificationsOpen = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElNotifications(anchorElNotifications ? null : event.currentTarget);
  };
  
  const handleFilterOpen = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElFilter(anchorElFilter ? null : event.currentTarget);
  };
```
Don't forget to separate logical things from each other by inserting empty line